lib/matplotlib/rcsetup.py

```diff
@@ -60,8 +62,28 @@ def __call__(self, s):
                          % (self.key, s, list(six.itervalues(self.valid))))
 
 
+def _listify_validator(scalar_validator):
+    def f(s):
+        if isinstance(s, six.string_types):
+            try:
+                return [scalar_validator(v.strip()) for v in s.split(',')
+                        if v.strip()]
+            except Exception:
+                # Sometimes, a list of colors might be a single string
+                # of single-letter colornames. So give that a shot.
+                return [scalar_validator(v.strip()) for v in s if v.strip()]
+        elif type(s) in (list, tuple):
+            return [scalar_validator(v) for v in s if v]
+        else:
+            msg = "'s' must be of type [ string | list | tuple ]"
+            raise ValueError(msg)
+    f.__doc__ = scalar_validator.__doc__
+    return f
+
```

Comment:
We don't have to worry about `None`, because this only applies to string parsing. There wouldn't be any `None` objects in the iteration of the `split()` results.

Allowing for the short-hand notation 'rgb' == ['r', 'g', 'b'] is to reduce surprises. If you were using the cycler directly, then this is perfectly legitimate. But prior to this fix, if you were to specify the identical construction in the rcparam, then it would fail. Don't worry about other colors that don't have single-letters. If you want to use them, then you have to specify the list notation anyway for both the rcparam construction and for direct cycler construction. The entire point is to allow similar notations in both places. This would also be helpful for some other properties like markers that are denoted by single characters.